chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/curesStatistics/email/CuresStatisticsChartSpreadsheet.java

```diff
@@ -0,0 +1,299 @@
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.B_7_CURES, B_7_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.B_8_CURES, B_8_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.B_9_CURES, B_9_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.B_10, B_10_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.C_3_CURES, C_3_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.D_2_CURES, D_2_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.D_3_CURES, D_3_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.D_10_CURES, D_10_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.D_12, D_12_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.D_13, D_13_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.E_1_CURES, E_1_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.F_5_CURES, F_5_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.G_6_CURES, G_6_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.G_9_CURES, G_9_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.G_10, G_10_ROW_IDX));
+    }
+
+    public File generateSpreadsheet(Map<CertificationCriterionDTO, CuresCriterionChartStatistic> dataMap, LocalDate reportDataDate) throws IOException {
+        File newFile = copyTemplateFileToTemporaryFile();
+        Workbook workbook = getWorkbook(newFile);
+
+        populateDataSheet(dataMap, workbook);
+
+        updateChartTitles(workbook, reportDataDate);
+
+        XSSFFormulaEvaluator.evaluateAllFormulaCells(workbook);
+
+        return writeFileToDisk(workbook, newFile);
+    }
+
+    private File writeFileToDisk(Workbook workbook, File saveFile) throws IOException {
+        try (FileOutputStream outputStream = new FileOutputStream(saveFile)) {
+            workbook.write(outputStream);
+            workbook.close();
+        }
+        return saveFile;
+    }
+
+    private void updateChartTitles(Workbook workbook, LocalDate reportDate) {
+        List<Sheet> allChartSheets = new ArrayList<Sheet>();
+        allChartSheets.add(workbook.getSheetAt(2));
+        allChartSheets.add(workbook.getSheetAt(3));
+        allChartSheets.add(workbook.getSheetAt(4));
```

Comment:
And seconds after my comment about loving removing magic numbers, Checkstyle calls out these... 🤣